core/query.go

```diff
@@ -0,0 +1,35 @@
+package lucy
+
+import lucyErr "Neo4j-OGM/errors"
+
+type QueryQueue struct {
+	queries *[]Query
+}
+
+func (q *QueryQueue) Init() {
+	queries := make([]Query, 0)
+	q.queries = &queries
+}
+
+func (q *QueryQueue) Push(query Query) {
+	*q.queries = append(*q.queries, query)
```

Comment:
Potentially dangerous dereference here

Reply:
The Init() function in the struct, initializes q.queries appropriately. There is no chance of a nil pointer dereference if Init() is called.